app/celery/provider_tasks.py

```diff
@@ -94,6 +96,18 @@ def deliver_letter(self, notification_id):
     notification = notifications_dao.get_notification_by_id(notification_id, _raise=True)
     postal_address = PostalAddress(notification.to, allow_international_letters=True)
 
+    if notification.status != NOTIFICATION_CREATED:
+        current_app.logger.warning(
+            "deliver_letter task called for notification %s in status %s", notification_id, notification.status
+        )
+        return
+
+    if notification.key_type != KEY_TYPE_NORMAL:
+        current_app.logger.warning(
```

Comment:
I might be tempted to log an error on this, so that it gets flagged in sentry and actually becomes visible. Otherwise is there a risk that we just emit warnings every day for the same notifications because their state doesn't change?

Reply:
added to this - not sure if it was obvious to you but i haven't changed  the status == created check since we'd expect that a letter not in created wouldnt get picked up again next time